contracts/staking/DonationsStaking.sol

```diff
@@ -0,0 +1,142 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity >=0.8.0;
+
+pragma experimental ABIEncoderV2;
+
+import "../Interfaces.sol";
+import "./SimpleStaking.sol";
+import "../utils/DAOUpgradeableContract.sol";
+
+/**
+ * @title DonationStaking contract that receives funds in ETH/DAI
+ * and stake them in the SimpleStaking contract
+ */
+contract DonationsStaking is DAOUpgradeableContract {
+	SimpleStaking public stakingContract;
+	cERC20 public DAI;
```

Comment:
- replace DAI with generic token
- add method setStakingContract(contract,token) onlyavatar - this method will call withdraw(), and re-approve new token
- replace all instances/names of dai with generic token

Reply:
Changed it as well